File: backend/src/main/java/isep/webapp/terasync/model/odSystem/ProductImage.java
package isep.webapp.terasync.model.odSystem;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Setter
@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class ProductImage extends isep.webapp.terasync.model.Entity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "productImageId")
    private int productImageId;
    @Column(name = "name")
    private String name;
    @Column(name = "description")
    private String description;
    @Column(name = "path")
    private String path;
    @Column(name = "productId")
    private int productId;
}

File: front/src/app/model/odSystem/product.image.ts
export class ProductImage {
  productImageId: number | undefined;
  name: string;
  path: string;
  productId: number;
  description: string | undefined;

  imageUrl: string | undefined;
  mightDelete: boolean = false;

  constructor(name: string, path: string, productId: number, description?: string, productImageId?: number) {
    this.productImageId = productImageId;
    this.name = name;
    this.description = description;
    this.path = path;
    this.productId = productId;
  }

  public static fromJson(json: ProductImage): ProductImage {
    return new ProductImage(json.name, json.path, json.productId,
      json.description, json.productImageId);
  }

  static initializeProductImages(json: {productImages: ProductImage[]}) {
    let productImages: ProductImage[] = [];
    if (json.productImages) {
      for (let productImage of json.productImages) {
        productImages.push(ProductImage.fromJson(productImage));
      }
    }
    return productImages;
  }
}
